website_product_configurator/static/src/js/config_form.esm.js

```diff
@@ -0,0 +1,629 @@
+/** @odoo-module **/
+
+import Dialog from "@web/legacy/js/core/dialog";
+import {insertThousandsSep} from "@web/core/utils/numbers";
+import {jsonrpc} from "@web/core/network/rpc_service";
+import {localization} from "@web/core/l10n/localization";
+import publicWidget from "@web/legacy/js/public/public_widget";
+
+publicWidget.registry.ProductConfigurator = publicWidget.Widget.extend({
+    selector: ".product_configurator",
+    events: {
+        "change.datetimepicker #product_config_form .input-group.date":
+            "_onChangeDateTime",
+        "change #product_config_form .config_attribute": "_onChangeConfigAttribute",
+        "change #product_config_form .custom_config_value.config_attachment":
+            "_onChangeFile",
+        "change #product_config_form .custom_config_value": "_onChangeCustomField",
+        "click #product_config_form .config_step": "_onClickConfigStep",
+        "click #product_config_form .btnNextStep": "_onClickBtnNext",
+        "click #product_config_form .btnPreviousStep": "_onClickBtnPrevious",
+        "submit #product_config_form": "_onSubmitConfigForm",
+        "click #product_config_form .image_config_attr_value_radio":
+            "_onClickRadioImage",
+        "change #product_config_form .custom_config_value.spinner_qty":
+            "_onChangeQtySpinner",
+        "click #product_config_form .js_add_qty": "_onClickAddQty",
+        "click #product_config_form .js_remove_qty": "_onClickRemoveQty",
+    },
+
+    init: function () {
+        this._super.apply(this, arguments);
+        this.config_form = $("#product_config_form");
+        // For file (custom field)
+        this.image_dict = {};
+    },
+
+    _onChangeConfigAttribute: function (event) {
+        var self = this;
+        var attribute = [event.currentTarget];
+        self._checkRequiredFields(attribute);
+        var flag = self._checkChange(self);
+        if (flag) {
+            var form_data = self.config_form.serializeArray();
+            for (var field_name in self.image_dict) {
+                form_data.push({
+                    name: field_name,
+                    value: self.image_dict[field_name],
+                });
+            }
+
+            self.call("ui", "block");
+            jsonrpc("/website_product_configurator/onchange", {
+                form_values: form_data,
+                field_name: $(attribute[0]).getAttributes().name,
+            }).then(function (data) {
```

Comment:
nitpicking / not-blocking: use of async functions

We could avoid many nested callbacks and thus improve readability
In combination with arrow functions, the use of `var self = this` becomes unnecessary, for example